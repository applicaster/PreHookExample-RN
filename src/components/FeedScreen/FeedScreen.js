import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Dimensions,
  FlatList,
  Modal,
  View,
  NativeModules,
  Platform,
} from 'react-native';
import EventContainer from '../EventContainer';
import MediaDetailsModal from '../MediaDetailsModal';
import WritePostModal from '../WritePostModal';
import CloseButton from '../CloseButton';
import WritePostButton from '../WritePostButton';

class FeedScreen extends Component {
  static navigationOptions = ({ screenProps }) => ({
    ...screenProps,
    headerLeft: <CloseButton onPress={FeedScreen.closeFeed} />,
  });

  static closeFeed() {
    return (Platform.OS === 'ios')
      ? NativeModules.ZPReactNativeBridgeListener.postEvent('dismiss_modal_view', { animated: 1 }, () => {})
      : NativeModules.APReactNativeBridge.handleCommand('stop', {});
  }

  constructor(props) {
    super(props);
    this.onRefresh = this.onRefresh.bind(this);
  }

  getChildContext() {
    return { navigation: this.props.navigation };
  }

  componentWillMount() {
    this.props.setAccountId('59b7a7caf422c00009d974f0');
    this.props.setTimelineId('59b7cf76044ab9298c61afa3');
    this.props.setTimezone('3600');
    this.props.setEnvironment('production');
    this.props.fetchSocialEvents();
  }

  onRefresh() {
    this.props.fetchSocialEvents();
  }

  render() {
    const { isFacebookAvailable, isTwitterAvailable, socialEvents, loading, toggleModal } = this.props;
    const backgroundFeedColor = { backgroundColor: this.context.backgroundColor };
    return (
      <View style={[backgroundFeedColor, { flex: 1 }]}>
        <FlatList
          data={socialEvents}
          renderItem={({item}) => <EventContainer key={item.id} event={item} />}
          keyExtractor={(item) => item.id }
          style={[backgroundFeedColor]} contentContainerStyle={[backgroundFeedColor]}
          refreshing={loading}
          onRefresh={this.onRefresh}
          initialNumToRender={4}
          onEndReached={() => {}}
          onEndReachedThreshold={1}
        />
        <MediaDetailsModal />
        <WritePostModal />
        <WritePostButton />
      </View>
    );
  }
}

FeedScreen.propTypes = {
  fetchSocialEvents: PropTypes.func,
  isFacebookAvailable: PropTypes.bool,
  isTwitterAvailable: PropTypes.bool,
  loading: PropTypes.bool,
  setAccountId: PropTypes.func,
  setTimelineId: PropTypes.func,
  setTimezone: PropTypes.func,
  setEnvironment: PropTypes.func,
  socialEvents: PropTypes.array,
  navigation: PropTypes.object,
};

FeedScreen.contextTypes = {
  backgroundColor: PropTypes.string,
};

FeedScreen.childContextTypes = {
  navigation: PropTypes.object,
};

export default FeedScreen;
