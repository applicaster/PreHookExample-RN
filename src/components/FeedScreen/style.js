import {
  StyleSheet,
} from 'react-native';

export const styles = StyleSheet.create({
  feedListContent: {
    backgroundColor: 'transparent',
    paddingTop: 5,
  },
  feedList: {
    backgroundColor: 'transparent',
    paddingBottom: 5,
  },
});
