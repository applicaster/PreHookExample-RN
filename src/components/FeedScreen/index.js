import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  fetchSocialEvents,
  toggleModal,
  updateFavoriteTweets,
} from '../../actions';

import {
  getSocialEvents,
  getLoading,
  isFacebookAvailable,
  isTwitterAvailable,
} from '../../selectors';

import FeedScreen from './FeedScreen';

const mapStateToProps = state => ({
  isFacebookAvailable: isFacebookAvailable(state),
  isTwitterAvailable: isTwitterAvailable(state),
  loading: getLoading(state),
  socialEvents: getSocialEvents(state),
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchSocialEvents,
  toggleModal,
  updateFavoriteTweets,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(FeedScreen);
