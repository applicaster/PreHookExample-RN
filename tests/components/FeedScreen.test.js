import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { event } from './testDataUtils';
import FeedScreen from '../../src/components/FeedScreen/FeedScreen';

jest.mock('react-native-zapp-bridge', () => ({
  sendAnalyticEvent: () => Promise.resolve(),
}));
jest.mock('@applicaster/feed-rn-utils', () => 'FeedRNUtils');
jest.mock('NativeEventEmitter', () => class MockNativeEventEmitter {
  addListener = () => jest.fn();
  removeListener = () => jest.fn();
  removeAllListeners = () => jest.fn();
});
jest.mock('../../src/components/EventContainer', () => 'EventContainer');
jest.mock('../../src/components/ModalScreen', () => 'ModalScreen');
jest.mock('../../src/components/WritePostButton', () => 'WritePostButton');


describe('FeedScreen', () => {
  test('component renders correctly without events', () => {
    const tree = renderer.create(
      <FeedScreen
        fetchEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading={false}
        socialEvents={[]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly with events', () => {
    const tree = renderer.create(
      <FeedScreen
        fetchEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading={false}
        socialEvents={[event(1), event(2)]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly while loading', () => {
    const tree = renderer.create(
      <FeedScreen
        fetchEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading
        socialEvents={[]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly while loading with events', () => {
    const tree = renderer.create(
      <FeedScreen
        fetchEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading
        socialEvents={[event(1), event(2)]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly with modal on top', () => {
    const tree = renderer.create(
      <FeedScreen
        fetchEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading
        socialEvents={[event(1), event(2)]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly with no write post button', () => {
    const tree = renderer.create(
      <FeedScreen
        fetchEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading
        socialEvents={[event(1), event(2)]}
        isFacebookAvailable={false}
        isTwitterAvailable={false}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly with write post button due to facebook availability', () => {
    const tree = renderer.create(
      <FeedScreen
        fetchEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading
        socialEvents={[event(1), event(2)]}
        isFacebookAvailable
        isTwitterAvailable={false}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly with write post button due to twitter availability', () => {
    const tree = renderer.create(
      <FeedScreen
        fetchEvents={() => {}}
        setAccountId={() => {}}
        setTimelineId={() => {}}
        setTimezone={() => {}}
        setEnvironment={() => {}}
        loading
        socialEvents={[event(1), event(2)]}
        isFacebookAvailable={false}
        isTwitterAvailable
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
