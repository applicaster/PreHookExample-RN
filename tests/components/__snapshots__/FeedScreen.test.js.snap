// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedScreen component renders correctly while loading 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "undefined99",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingTop": 5,
        },
      ]
    }
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={4}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        progressViewOffset={undefined}
        refreshing={false}
      />
    }
    refreshing={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingBottom": 5,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <ModalScreen />
  <WritePostButton />
</View>
`;

exports[`FeedScreen component renders correctly while loading with events 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "undefined99",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingTop": 5,
        },
      ]
    }
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={4}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        progressViewOffset={undefined}
        refreshing={false}
      />
    }
    refreshing={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingBottom": 5,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <ModalScreen />
  <WritePostButton />
</View>
`;

exports[`FeedScreen component renders correctly with events 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "undefined99",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingTop": 5,
        },
      ]
    }
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={4}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        progressViewOffset={undefined}
        refreshing={false}
      />
    }
    refreshing={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingBottom": 5,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <ModalScreen />
  <WritePostButton />
</View>
`;

exports[`FeedScreen component renders correctly with modal on top 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "undefined99",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingTop": 5,
        },
      ]
    }
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={4}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        progressViewOffset={undefined}
        refreshing={false}
      />
    }
    refreshing={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingBottom": 5,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <ModalScreen />
  <WritePostButton />
</View>
`;

exports[`FeedScreen component renders correctly with no write post button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "undefined99",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingTop": 5,
        },
      ]
    }
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={4}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        progressViewOffset={undefined}
        refreshing={false}
      />
    }
    refreshing={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingBottom": 5,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <ModalScreen />
  <WritePostButton />
</View>
`;

exports[`FeedScreen component renders correctly with write post button due to facebook availability 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "undefined99",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingTop": 5,
        },
      ]
    }
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={4}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        progressViewOffset={undefined}
        refreshing={false}
      />
    }
    refreshing={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingBottom": 5,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <ModalScreen />
  <WritePostButton />
</View>
`;

exports[`FeedScreen component renders correctly with write post button due to twitter availability 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "undefined99",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingTop": 5,
        },
      ]
    }
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={4}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        progressViewOffset={undefined}
        refreshing={false}
      />
    }
    refreshing={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingBottom": 5,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <ModalScreen />
  <WritePostButton />
</View>
`;

exports[`FeedScreen component renders correctly without events 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "undefined99",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingTop": 5,
        },
      ]
    }
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={4}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        progressViewOffset={undefined}
        refreshing={false}
      />
    }
    refreshing={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "paddingBottom": 5,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
  <ModalScreen />
  <WritePostButton />
</View>
`;
